### PR TITLE
fix: Upload controlled issue

### LINKED ISSUE #26427: upload 组件在 form 下使用 customRequest 方法，onChange 只触发一次
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/upload-onchange-wdspg?file=/index.js)

### Steps to reproduce
打开链接，点击上传，查看 console 打印

### What is expected?
onChange 被完整调用

### What is actually happening?
onChange 只调用一次

| Environment | Info |
|---|---|
| antd | 4.3.1 |
| React | 16.8.6 |
| System | macOS |
| Browser | Chrome 84.0.4147.135 |

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

resolve #26427

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Fix Upload list status issue when in control mode.       |
| 🇨🇳 Chinese |   修复 Upload 在受控模式下同步更新导致的状态错误问题。       |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by xrkffgg** [comment]
snap 好像需要更新

**Comment by xrkffgg** [comment]
image

**Comment by zombieJ** [comment]
奇怪，本地测试是过的。我更新一下试试。

File: components/upload/Upload.tsx [hunk 1]
```diff
@@ -62,15 +64,15 @@ const InternalUpload: React.ForwardRefRenderFunction<unknown, UploadProps> = (pr
...
-    if ('fileList' in props) {
-      setFileList(fileListProp || []);
+    if (fileListProp !== undefined && fileListProp !== fileListRef.current) {
+      fileListRef.current = fileListProp;
+      forceUpdate();
     }
   }, [fileListProp]);
 
   const onChange = (info: UploadChangeParam) => {
-    if (!('fileList' in props)) {
```
Comment (Kermit-Xuan) [inline components/upload/Upload.tsx:71]: 是不是核心问题就出在这，受控模式下，没有调用 `setFileList` 同步状态

File: components/upload/__tests__/__snapshots__/uploadlist.test.js.snap [hunk 2]
```diff
@@ -28,7 +28,7 @@ exports[`Upload List handle error 1`] = `
     class="ant-upload-list ant-upload-list-text"
   >
     <div
-      class="ant-upload-animate-enter"
+      class=""
```
Comment (afc163) [inline components/upload/__tests__/__snapshots__/uploadlist.test.js.snap:31]: 检查看看动画是不是有影响

Comment (zombieJ) [reply components/upload/__tests__/__snapshots__/uploadlist.test.js.snap:31]: 动画有的

File: components/upload/Upload.tsx (hunk 1 above)
Comment (zombieJ) [reply components/upload/Upload.tsx:71]: 一半一半，这边 `setFileList` 在受控下，第二次上传还是会卡住。主要还是过去实现的不好（当然现在也是个 patch），让 fileList 即管状态又管展示。同步下设置会导致重置回当前 props 的问题。

**Pull request merged**